src/react-components/chat-message.js

```diff
@@ -60,38 +94,123 @@ export function spawnChatMessage(body) {
         [styles.presenceLogEmoji]: isEmoji
       })}
     >
-      {messageBodyDom(body)}
+      {messageBodyDom(body, from)}
     </div>
   );
 
-  ReactDOM.render(entryDom, el, () => {
-    // Scale by 12x
-    messageCanvas.width = el.offsetWidth * 12.1;
-    messageCanvas.height = el.offsetHeight * 12.1;
-
-    const xhtml = encodeURIComponent(`
-        <svg xmlns="http://www.w3.org/2000/svg" width="${messageCanvas.width}" height="${messageCanvas.height}">
-          <foreignObject width="8.333%" height="8.333%" style="transform: scale(12.0);">
-            ${serializeElement(el)}
-          </foreignObject>
-        </svg>
-  `);
-    const img = new Image();
-
-    img.onload = async () => {
-      context.drawImage(img, 0, 0);
-      const blob = await new Promise(resolve => messageCanvas.toBlob(resolve));
-      document.querySelector("a-scene").emit("add_media", new File([blob], "message.png", { type: "image/png" }));
-      el.parentNode.removeChild(el);
-    };
-
-    img.src = "data:image/svg+xml," + xhtml;
+  return new Promise(resolve => {
+    ReactDOM.render(entryDom, el, () => {
+      // Scale by 12x
+      let objectScale = "8.33%";
+      let scale = 12;
+
+      if (lowResolution) {
+        // In low res, scale by 4x
+        objectScale = "25%";
+        scale = 4;
+      }
+
+      messageCanvas.width = el.offsetWidth * (scale + 0.1);
+      messageCanvas.height = el.offsetHeight * (scale + 0.1);
+
+      const xhtml = encodeURIComponent(`
+          <svg xmlns="http://www.w3.org/2000/svg" width="${messageCanvas.width}" height="${messageCanvas.height}">
+            <foreignObject width="${objectScale}" height="${objectScale}" style="transform: scale(${scale});">
+              ${serializeElement(el)}
+            </foreignObject>
+          </svg>
+    `);
+      const img = new Image();
+
+      img.onload = async () => {
+        context.drawImage(img, 0, 0);
+        const blob = await new Promise(resolve => messageCanvas.toBlob(resolve));
+        el.parentNode.removeChild(el);
+        resolve(blob);
+      };
+
+      img.src = "data:image/svg+xml," + xhtml;
+    });
   });
 }
 
-export default function ChatMessage(props) {
-  const isOneLine = props.body.split("\n").length === 1;
+export async function createInWorldLogMessage({ name, type, body }) {
```

Comment:
Would like us to revisit this so that incoming messages don't render past message unreadable. I spent some time playing with the parameters of the animations configured here. I suggest at least changing the parameters such that 1) the messages disappear sooner (15 seconds is too long imo) and 2) they translate more aggressively (so that they are spread out a bit more). This will not fix the underlying problem (which is that new messages hide old ones, but is easy to do and makes it less likely to happen.

Reply:
i sped them up, but increasing the distances they travel just strikes me as past the annoying threshold. i really think there's not going to be a way to show more than one message in your visual field without some kind of modality -- the amount of space needed (and the amount of motion needed to put on the chat messages) just for me personally isn't the right tradeoff. i think hover actions like you suggested are the way to go.